File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.cpp
class Solution {
public:
    string sortVowels(string s) {
        vector<int> Eindex;
        unordered_map<int, bool> mp;
        int sz = s.length();
        for(int i=0; i<sz; i++){
            char tmp = tolower(s[i]);
            if(tmp == 'a' || tmp == 'e' || tmp == 'i' || tmp == 'o' || tmp == 'u'){
                mp[i] = true;
                Eindex.push_back(i);
            } 
        }

        sort(Eindex.begin(), Eindex.end(), [&](int a, int b){
            return s[a] < s[b];
        });

        string ans = "";
        int ind = 0;
        for(int i=0; i<sz; i++){
            if(mp.find(i) != mp.end()){
                ans += s[Eindex[ind++]];
            }
            else ans += s[i];
        }

        return ans;
    }
};

File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.java
class Solution {
    public String sortVowels(String s) {
        int sz = s.length();
        char ch = ' ';
        List<Integer> inds = new ArrayList<>();
        List<Character> vowels = new ArrayList<>(), req = new ArrayList<>(Arrays.asList('a','e','i','o','u','A','E','I','O','U'));
        StringBuilder ans = new StringBuilder(s);

        for(int i=0; i<sz; i++){
            ch = s.charAt(i);
            if(req.indexOf(ch) > -1){
                vowels.add(s.charAt(i));
                inds.add(i);
            }
        }

        Collections.sort(vowels, (a, b) -> {
            return (int) a - b;
        });

        int k = 0;
        for(int i : inds){
            ans.setCharAt(i, vowels.get(k++));
        }

        return ans.toString();
    }
}